inference-engine/src/vpu/graph_transformer/src/stages/ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,163 @@
+        return std::make_shared<CTCGreedyDecoderSeqLenStage>(*this);
+    }
+
+    void finalizeDataLayoutImpl() override {
+    }
+
+    void getBatchSupportInfoImpl(StageDataInfo<BatchSupport>&) override {
+    }
+
+    void propagateDataOrderImpl(StageDataInfo<DimsOrder>& orderInfo) override {
+        auto input0 = inputEdge(0)->input();
+        auto input1 = inputEdge(1)->input();
+        auto output0 = outputEdge(0)->output();
+        auto output1 = outputEdge(1)->output();
+
+        orderInfo.setInput(inputEdge(0), DimsOrder::fromNumDims(input0->desc().numDims()));
+        orderInfo.setInput(inputEdge(1), DimsOrder::fromNumDims(input1->desc().numDims()));
+        orderInfo.setOutput(outputEdge(0), DimsOrder::fromNumDims(output0->desc().numDims()));
+        orderInfo.setOutput(outputEdge(1), DimsOrder::fromNumDims(output1->desc().numDims()));
+
+        if (numInputs() == 3) {
+            auto input2 = inputEdge(2)->input();
+            orderInfo.setInput(inputEdge(2), DimsOrder::fromNumDims(input2->desc().numDims()));
+        }
+    }
+
+    void getDataStridesRequirementsImpl(StageDataInfo<StridesRequirement>& stridesInfo) override {
+        stridesInfo.setInput(inputEdge(0), StridesRequirement::compact());
+        stridesInfo.setInput(inputEdge(1), StridesRequirement::compact());
+        if (numInputs() == 3) {
+            stridesInfo.setInput(inputEdge(2), StridesRequirement::compact());
+        }
+        stridesInfo.setOutput(outputEdge(0), StridesRequirement::compact());
+        stridesInfo.setOutput(outputEdge(1), StridesRequirement::compact());
+    }
+
+    StageSHAVEsRequirements getSHAVEsRequirementsImpl() const override {
+        return StageSHAVEsRequirements::OnlyOne;
+    }
+
+    void initialCheckImpl() const override {
+        VPU_THROW_UNLESS(numInputs() == 2 || numInputs() == 3,
+                         "{} stage with name {} must have 2 or 3 inputs, actually "
+                         "provided {} inputs",
+                         type(), name(), numInputs());
+        VPU_THROW_UNLESS(numOutputs() == 2,
+                         "{} stage with name {} must have 2 output, actually "
+                         "provided {} outputs",
+                         type(), name(), numOutputs());
+
+        if (numInputs() == 2) {
+            assertInputsOutputsTypes(
+                this, {{DataType::FP16}, {DataType::S32}},
+                {{DataType::S32}, {DataType::S32}});
+        } else {
+            assertInputsOutputsTypes(
+                this, {{DataType::FP16}, {DataType::S32}, {DataType::S32}},
+                {{DataType::S32}, {DataType::S32}});
+        }
+    }
+
+    void serializeParamsImpl(BlobSerializer& serializer) const override {
+        const auto mergeRepeated = attrs().get<bool>("mergeRepeated");
+
+        const auto blankIndex = [&] {
+            if (numInputs() == 3) {
+                const auto& input = inputEdge(2)->input()->content()->get<int32_t>();
```

Comment:
3d input can be only `Const`? If yes, please, add assert on this.
Also, we need a check for a total size of 3d input (as I can see it must be 1)

Reply:
There is no restriction on const-ness of the third input in the documentation. 
Yes, size should be always == 1.

Reply:
I see that there is no restriction in the docs. But on VPU side, as I can see, we can handle only const.
I suggest checking it in the vpu frontend parser

Reply:
BTW, we can do extraction of `blankIndex` in `parseCTCGreedyDecoderSeqLen` and pass it as stage attribute

Reply:
Moved blankIndex to parseCTC.
I don't get why 3rd output must be a const. parseCTC get `DataVector inputs`, I don't see any reason why we should restrict it to only const values.

Reply:
Network constants have been added to the model before any stage is parsed.
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/vpu/graph_transformer/src/frontend/parse_data.cpp#L117
`DataContent::_content` is set with valid pointer only in case when you do `addConstData`. In other cases, it is still nullptr.
And if you do not assert that 3d input has `DataUsage::Const`, you can dereference nullptr while doing `->get<int32_t>();`

Reply:
Somehow it completely went over my head that I try to get a value of an input tensor at the layer parsing stage. Maybe the whole "conversion from an input to the layer parameter" isn't a good idea. What's the point of having an input with size 1, if it could be easily a layer parameter. It was probably intended to be able to set the `blankIndex` dynamically.

Reply:
@Maxim-Doronin At the end of the day, I moved 3rd input to the param list again. It had some weird issues and most likely also related to the {1,1,1} tensor bug thing. Changes are reverted, please review.